proto/src/error.rs

```diff
@@ -1,51 +1,24 @@
 use miden_crypto::merkle::MmrError;
+use thiserror::Error;
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Error, Debug, PartialEq)]
 pub enum ParseError {
```

Comment:
Question: do we not need to implement `Clone` if we implement `Error`?

Reply:
We need. I missed it, will revert back.